src/gate_detect/include/Gate.h

```diff
@@ -0,0 +1,117 @@
+//
+// Created by da-cam on 20/02/18.
+//
+
+#ifndef PROJECT_GATE_H
+#define PROJECT_GATE_H
+
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+class Gate {
+  public:
+    Gate();
+
+    /*
+     * Function to be called on cv::Mat to find vertical poles and horizontal
+     * poles
+     *
+     * mat : cv::Mat potentially containing a gate
+     *
+     */
+
+    std::vector<float> initialize(const cv::Mat matin);
+
+    /*
+     * Function to check if Gate object has been initialized
+     */
+
+    bool checkMat();
+
+    void vertLineCheck();
+
+    void horLineCheck();
+
+  private:
+    cv::Mat dst;
+    int lowVertThresh, lowHorThresh, cannyLow, cannyHigh, poleMax,
+    camera_res_width;
+    float m, b;
```

Comment:
needs better names, what are these?

Reply:
Entered detailed descriptions for HoughLinesP and Canny filter